xapian-core/tests/api_closedb.cc

```diff
@@ -484,6 +484,17 @@ DEFINE_TESTCASE(closedb10, writable && metadata) {
 		   db.get_metadata("bar"));
     TEST_EXCEPTION(Xapian::DatabaseClosedError,
 		   db.metadata_keys_begin());
+	return true;
+}
 
-    return true;
+DEFINE_TESTCASE(closedb11, remote) {
+	Xapian::WritableDatabase db(get_writable_database());
+	kill_server(db.get_uuid());
```

Comment:
I think the UUID of the database is not a good way to specify which server to kill.  The same database can be open multiple times for reading, as well as potentially once for writing.  While the UU stands for "Universally Unique", that just means each database on disk should have a different UUID - if you open the same database twice the UUID won't be unique, and if the specified UUID maps to more than one open database then we won't know which server process to kill.

Passing `kill_server()` a pointer to the `Database`/`WritableDatabase` object might work.  At least for a single-shard database it can just read the internal pointer and use that to find the process id.